samples/snippets/schema.py

```diff
@@ -0,0 +1,489 @@
+    # [START pubsub_get_schema]
+    from google.api_core.exceptions import NotFound
+    from google.cloud.pubsub import SchemaServiceClient
+
+    # TODO(developer): Replace these variables before running the sample.
+    # project_id = "your-project-id"
+    # schema_id = "your-schema-id"
+
+    schema_client = SchemaServiceClient()
+    schema_path = schema_client.schema_path(project_id, schema_id)
+
+    try:
+        result = schema_client.get_schema(request={"name": schema_path})
+        print(f"Got a schema:\n{result}")
+    except NotFound:
+        print(f"{schema_id} not found.")
+    # [END pubsub_get_schema]
+
+
+def list_schemas(project_id):
+    """List schema resources."""
+    # [START pubsub_list_schemas]
+    from google.cloud.pubsub import SchemaServiceClient
+
+    # TODO(developer): Replace these variables before running the sample.
+    # project_id = "your-project-id"
+
+    project_path = f"projects/{project_id}"
+    schema_client = SchemaServiceClient()
+
+    for schema in schema_client.list_schemas(request={"parent": project_path}):
+        print(schema)
+
+    print("Listed schemas.")
+    # [END pubsub_list_schemas]
+
+
+def delete_schema(project_id, schema_id):
+    """Delete a schema resource."""
+    # [START pubsub_delete_schema]
+    from google.api_core.exceptions import NotFound
+    from google.cloud.pubsub import SchemaServiceClient
+
+    # TODO(developer): Replace these variables before running the sample.
+    # project_id = "your-project-id"
+    # schema_id = "your-schema-id"
+
+    schema_client = SchemaServiceClient()
+    schema_path = schema_client.schema_path(project_id, schema_id)
+
+    try:
+        schema_client.delete_schema(request={"name": schema_path})
+        print(f"Deleted a schema:\n{schema_path}")
+    except NotFound:
+        print(f"{schema_id} not found.")
+    # [END pubsub_delete_schema]
+
+
+def create_topic_with_schema(project_id, topic_id, schema_id, message_encoding):
+    """Create a topic resource with a schema."""
+    # [START pubsub_create_topic_with_schema]
+    from google.api_core.exceptions import AlreadyExists
+    from google.cloud.pubsub import PublisherClient, SchemaServiceClient
+    from google.pubsub_v1.types import Encoding
+
+    # TODO(developer): Replace these variables before running the sample.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+    # schema_id = "your-schema-id"
+    # Choose either BINARY or JSON as valid message encoding in this topic.
+    # message_encoding = "BINARY"
+
+    publisher_client = PublisherClient()
+    topic_path = publisher_client.topic_path(project_id, topic_id)
+
+    schema_client = SchemaServiceClient()
+    schema_path = schema_client.schema_path(project_id, schema_id)
+
+    if message_encoding == "BINARY":
+        encoding = Encoding.BINARY
+    elif message_encoding == "JSON":
+        encoding = Encoding.JSON
+    else:
+        encoding = Encoding.ENCODING_UNSPECIFIED
```

Comment:
Any reason to not use the `types.Encoding` enum directly?

If we use string constants in the sample, a good portion of users will follow suit, but that requires some extra boilerplate code and is also susceptible to typos.

Reply:
The main reason I'm writing this way is because I'm not sure how to pass `types.Encoding.BINARY` in place of `BINARY` in the following command: 

`python schema.py $PROJECT_ID publish-avro $TOPIC_ID resources/us-states.avsc BINARY`

Reply:
Ah, this can come through cmd arguments, fine then. I like the improvement that restricts the valid argument choices to "BINARY" or "JSON".